### PR TITLE
Use `rest_is_field_included` function in REST API controllers.

### LINKED ISSUE #8290: Use `rest_is_field_included` function in REST API controllers.
## Feature Description

Use the `rest_is_field_included` function in rest api controller classes.

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Use `rest_is_field_included` in all of our custom endpoints, making sure that all endpoints have item schema setup. 

### QA

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #8290

### CODE REVIEW

File: includes/REST_API/Embed_Controller.php [hunk 1]
```diff
@@ -324,6 +331,75 @@ private function get_data_from_document( $html ) {
...
+		// Wrap the data in a response object.
+		$response = rest_ensure_response( $data );
+
+		return $response;
+	}
+
+
+	/**
+	 * Retrieves the link's schema, conforming to JSON Schema.
+	 *
```
Comment (swissspidy) [inline includes/REST_API/Embed_Controller.php:372]: Missing `@since`

File: includes/REST_API/Status_Check_Controller.php [hunk 2]
```diff
@@ -103,7 +103,68 @@ public function status_check( $request ) {
...
+			return $this->add_additional_fields_schema( $this->schema );
+		}
+
+		$schema = [
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'status',
+			'type'       => 'object',
+			'properties' => [
+				'success' => [
+					'description' => __( 'Boolean if status was success', 'web-stories' ),
```
Comment (swissspidy) [inline includes/REST_API/Status_Check_Controller.php]:
```suggestion
					'description' => __( 'Whether check was successful', 'web-stories' ),
```

**Review by swissspidy — APPROVED** [review]
Sweet!

File: includes/REST_API/Embed_Controller.php (hunk 1 above)
Comment (spacedmonkey) [inline includes/REST_API/Embed_Controller.php:372]:
```suggestion
	 *
	 * @since 1.10.0
	 *
```

**Pull request merged**